sensor/common/sensor/central_communication_impl.go

```diff
@@ -242,7 +247,41 @@ func (s *centralCommunicationImpl) initialSync(stream central.SensorService_Comm
 		return err
 	}
 
-	return s.initialPolicySync(stream, detector)
+	if err := s.initialPolicySync(stream, detector); err != nil {
+		return err
+	}
+
+	if s.clientReconciliation {
+		// If client is reconciling, Sensor needs to wait for Central to send the hashes
+		return s.initialDeduperSync(stream, s.syncTimeout)
+	}
+	return nil
+}
+
+func (s *centralCommunicationImpl) initialDeduperSync(stream central.SensorService_CommunicateClient, timeout time.Duration) error {
+	done := make(chan struct{})
+	var err error
+	var msg *central.MsgToSensor
+	go func() {
+		msg, err = stream.Recv()
+		close(done)
+	}()
+	select {
+	case <-time.After(timeout):
+		log.Error("timeout reached waiting for the initial deduper sync")
+		return errors.Errorf("%v timeout reached waiting for the initial deduper state", timeout)
```

Comment:
The problem with things at this state (without #7755) is that if this case is reached, we get into a fail loop. Sensor will restart the connection here and it will expect for the deduper state (again) and return an error (again).